Source/AwsCommonRuntimeKit/crt/Future.swift

```diff
@@ -91,4 +91,36 @@ extension Future {
 
         return promise
     }
+
+    static func whenAllComplete(_ futures: [Future<Value>]) -> Future<Void> {
```

Comment:
Should this be `Future<Value>` instead of `Future<Void>`?

Reply:
Seems like you'd want the value if the bundled futures produce one.

Reply:
ya I thought about returning each value in a completion as they come in, but I wrote this for a specific use case. I'll make it more generic :) good point

Reply:
leaving off for now and add documentation. will return to this if required